gwpy/timeseries/timeseries.py

```diff
@@ -1463,18 +1474,26 @@ def taper(self, side='leftright'):
         if side not in ('left', 'right', 'leftright'):
             raise ValueError("side must be one of 'left', 'right', "
                              "or 'leftright'")
+        if duration and nsamples:
```

Comment:
@tjma12, if the other suggestions don't appease Codeclimate, then consider removing this `ValueError` exception and just warning the user in the docstring that if both arguments are passed then `nsamples` will override `duration`, which is true under the changes I suggested.

Reply:
I agree, this can be achieved in the documentation.

Reply:
Removed this check in new commit